src/Interpreters/InterpreterSelectQueryAnalyzer.h

```diff
@@ -71,6 +71,13 @@ class InterpreterSelectQueryAnalyzer : public IInterpreter
     /// Set number_of_current_replica and count_participating_replicas in client_info
     void setProperClientInfo(size_t replica_number, size_t count_participating_replicas);
 
+    const QueryTreeNodePtr & getQueryTree() const { return query_tree; }
+
+    SelectQueryInfo getSelectQueryInfo()
```

Comment:
We should not introduce such method, because client can thing that this select query info can be used somewhere, but each table expression get unique select query info.
If this is only necessary for tests, maybe we can find some workaround ?

Reply:
Added friend struct from unit tests